### PR TITLE
PMM-10520 mysqld_exporter and node_exporter not support --log.level=fatal

### PR DESCRIPTION
**Pull request by YaroslavPodorvanov**
PMM-10520

Build: SUBMODULES-2720

| Exporter                                                                    | Logger                               | Usage                    | Support fatal |
|-----------------------------------------------------------------------------|--------------------------------------|--------------------------|---------------|
| [azure_metrics_exporter](https://github.com/percona/azure_metrics_exporter) | github.com/prometheus/common/log     | log.AddFlags(...)        | ✔️             |
| [mongodb_exporter](https://github.com/percona/mongodb_exporter)             | github.com/sirupsen/logrus           | log.SetLevel(...)        | ✔             |
| [mysqld_exporter](https://github.com/percona/mysqld_exporter)               | github.com/prometheus/common/promlog | flag.AddFlags(...)       | ❌             |
| [node_exporter](https://github.com/percona/node_exporter)                   | github.com/prometheus/common/promlog | flag.AddFlags(...)       | ❌             |
| [postgres_exporter](https://github.com/percona/postgres_exporter)           | github.com/prometheus/common/log     | log.AddFlags(...)        | ✔             |
| [proxysql_exporter](https://github.com/percona/proxysql_exporter)           | github.com/prometheus/common/log     | log.Base().SetLevel(...) | ✔             |
| [rds_exporter](https://github.com/percona/rds_exporter)                     | github.com/prometheus/common/log     | log.AddFlags(...)        | ✔             |

### CODE REVIEW

**Review by BupycHuk — APPROVED** [review]

File: admin/commands/inventory/add_agent.go [hunk 1]
```diff
@@ -23,7 +23,11 @@ var (
...
 
-func addExporterGlobalFlags(cmd *kingpin.CmdClause) {
+func addExporterGlobalFlags(cmd *kingpin.CmdClause, supportLogLevelFatal bool) {
 	// Add command global flags
-	cmd.Flag("log-level", "Service logging level").Default("warn").EnumVar(&addExporterLogLevel, "debug", "info", "warn", "error", "fatal")
+	if supportLogLevelFatal {
+		cmd.Flag("log-level", "Service logging level").Default("warn").EnumVar(&addExporterLogLevel, "debug", "info", "warn", "error", "fatal")
+	} else {
+		cmd.Flag("log-level", "Service logging level").Default("warn").EnumVar(&addExporterLogLevel, "debug", "info", "warn", "error")
+	}
```
Comment (artemgavrilov) [inline admin/commands/inventory/add_agent.go:32]:
```suggestion
	logLevels := []string{"debug", "info", "warn", "error"}
	if supportLogLevelFatal {
		logLevels = append(logLevels, "fatal")
	}
	
	cmd.Flag("log-level", "Service logging level").Default("warn").EnumVar(&addExporterLogLevel, logLevels...)
```

File: admin/commands/management/add.go [hunk 2]
```diff
@@ -31,12 +31,17 @@ var (
...
 	cmd.Flag("service-name", "Service name (overrides positional argument)").PlaceHolder("NAME").StringVar(&addServiceNameFlag)
 	cmd.Flag("host", "Service hostname or IP address (overrides positional argument)").StringVar(&addHostFlag)
 	cmd.Flag("port", "Service port number (overrides positional argument)").Uint16Var(&addPortFlag)
-	cmd.Flag("log-level", "Service logging level. One of: [debug, info, warn, error, fatal]").Default("warn").EnumVar(&addLogLevel, "debug", "info", "warn", "error", "fatal")
+
+	if supportLogLevelFatal {
+		cmd.Flag("log-level", "Service logging level. One of: [debug, info, warn, error, fatal]").Default("warn").EnumVar(&addLogLevel, "debug", "info", "warn", "error", "fatal")
+	} else {
+		cmd.Flag("log-level", "Service logging level. One of: [debug, info, warn, error]").Default("warn").EnumVar(&addLogLevel, "debug", "info", "warn", "error")
+	}
```
Comment (artemgavrilov) [inline admin/commands/management/add.go:44]: May be same as suggested above

File: managed/services/agents/log_level.go [hunk 3]
```diff
@@ -25,9 +25,15 @@ import (
...
-func withLogLevel(args []string, logLevel *string, pmmAgentVersion *version.Parsed) []string {
-	if pointer.GetString(logLevel) != "" && !pmmAgentVersion.Less(exporterLogLevelCommandVersion) {
-		args = append(args, "--log.level="+pointer.GetString(logLevel))
+func withLogLevel(args []string, logLevel *string, pmmAgentVersion *version.Parsed, supportLogLevelFatal bool) []string {
+	level := pointer.GetString(logLevel)
+
+	if level != "" && !pmmAgentVersion.Less(exporterLogLevelCommandVersion) {
+		if !supportLogLevelFatal && level == "fatal" {
+			level = "error"
+		}
```
Comment (artemgavrilov) [inline managed/services/agents/log_level.go:34]: It's need to fix problem for already added exporters with `fatal` level? Can you please add comment with description why it's needed.

**Review by artemgavrilov — APPROVED** [review]

File: managed/services/agents/log_level.go (hunk 3 above)
Comment (YaroslavPodorvanov) [reply managed/services/agents/log_level.go:34]: > It's need to fix problem for already added exporters with `fatal` level?

Yes, it's need to fix problem for already added exporters with `fatal` level

> Can you please add comment with description why it's needed.

Already in variable name `!supportLogLevelFatal`

File: admin/commands/inventory/add_agent.go (hunk 1 above)
Comment (YaroslavPodorvanov) [reply admin/commands/inventory/add_agent.go:32]: I prefer my solution as easier to read

File: admin/commands/management/add.go (hunk 2 above)
Comment (YaroslavPodorvanov) [reply admin/commands/management/add.go:44]: I prefer my solution as easier to read

File: managed/services/agents/log_level.go (hunk 3 above)
Comment (artemgavrilov) [reply managed/services/agents/log_level.go:34]: It's not so obvious why we we can face situation where fatal level is not supported AND we got fatal level. And as you confirmed it's possible for previously added exporters.

Comment (YaroslavPodorvanov) [reply managed/services/agents/log_level.go:36]: @artemgavrilov i just now add comment

**Review by michal-kralik — APPROVED** [review]

**Pull request merged**